main/views.py

```diff
@@ -19,3 +27,49 @@ def form_valid(self, form):
         data['id_list'] = new_id_list
         form.data = data
         return self.render_to_response(self.get_context_data(form=form))
+
+
+def upload_file(request):
+    if request.method == 'POST':
+        form = UploadFileForm(request.POST, request.FILES)
+        if form.is_valid():
+            return handle_uploaded_file(request.FILES['file'])
+    else:
+        form = UploadFileForm()
+    return render(request, 'upload.html', {'form': form})
+
+
+def handle_uploaded_file(f):
+    api = API(settings.USOS_URL, settings.USOS_CONSUMER_KEY,
+              settings.USOS_CONSUMER_SECRET)
+    with open('tmp/file.pdf', 'wb+') as destination:
```

Comment:
Please use some kind of temporary files (storing them in tmpfs, or, even better, using Python's tempfile module) to avoid saving data to disk. Also, the files does not seem to be deleted right now.